lib/src/code_field/code_controller.dart

```diff
@@ -27,20 +26,17 @@ class CodeController extends TextEditingController {
   Mode? _language;
 
   /// A highlight language to parse the text with
+  ///
+  /// Setting a language will change the analyzer to DefaultAnalyzer.
   Mode? get language => _language;
 
   set language(Mode? language) {
-    if (language == _language) {
-      return;
-    }
-
     if (language != null) {
-      _languageId = language.hashCode.toString();
-      highlight.registerLanguage(_languageId, language);
+      setLanguage(language, analyzer: const DefaultLocalAnalyzer());
+      return;
     }
 
     _language = language;
-    autocompleter.mode = language;
     _updateCode(_code.text);
     notifyListeners();
```

Comment:
Move everything to `setLanguage` so that if we want to deprecate and remove `set language` we can do this with the fewest of lines.